maven/bnd-testing-maven-plugin/src/main/java/aQute/bnd/maven/testing/plugin/TestingMojo.java

```diff
@@ -85,112 +81,42 @@ public void execute() throws MojoExecutionException, MojoFailureException {
 		if (!runFile.exists()) {
-			throw new MojoExecutionException("Could not find bnd run file " + runFile);
+			logger.error("Could not find bnd run file {}", runFile);
+			errors.incrementAndGet();
+			return;
 		}
-		try (StandaloneRun run = new StandaloneRun(runFile)) {
+		try (Bndrun run = new StandaloneBndrun(runFile)) {
 			Workspace workspace = run.getWorkspace();
 			workspace.setOffline(session.getSettings().isOffline());
 			for (RepositoryPlugin repo : workspace.getRepositories()) {
 				repo.list(null);
 			}
-			run.check();
-			if (!run.isOk()) {
-				throw new MojoExecutionException("Initializing the workspace failed " + run.getErrors());
+			run.report(errors);
+			if (errors.get() > 0) {
+				return;
 			}
-
 			if (resolve) {
-				resolve(run);
-			}
-			run.setProperty("-runbundles.tester", "biz.aQute.tester");
-			testing(run);
-		}
-	}
-
-	private void testing(Run run) throws Exception {
-		try {
-			System.out.println("Test " + run);
-			ProjectTester projectTester = run.getProjectTester();
-			File dir = new File(targetDir, "tmp");
-			IO.delete(dir);
-			dir.mkdirs();
-
-			projectTester.setCwd(dir);
-			projectTester.setReportDir(targetDir);
-			projectTester.test();
-
-			if (!run.getErrors().isEmpty()) {
-				System.out.println("Errors");
-				System.out.println(Strings.join("\n", run.getErrors()));
-				System.out.println();
-			}
-			if (!run.getWarnings().isEmpty()) {
-				System.out.println("Warnings");
-				System.out.println(Strings.join("\n", run.getWarnings()));
-				System.out.println();
-			}
-
-			IO.delete(dir);
-
-			if (run.getErrors().isEmpty())
-				return;
-
-			System.out.println(Strings.join("\n", run.getErrors()));
-			throw new MojoExecutionException("Test had errors");
-		} catch (Exception e) {
-			throw new MojoExecutionException(e.getMessage());
-		}
-	}
-
-	private void resolve(StandaloneRun run) throws Exception, IOException {
-		try (ProjectResolver projectResolver = new ProjectResolver(run)) {
-			List<Container> runBundles = projectResolver.getRunBundles();
-			Collection<Container> currentRunBundles = run.getRunbundles();
-			Collections.sort(runBundles, new Comparator<Container>() {
-
-				@Override
-				public int compare(Container o1, Container o2) {
-					return o1.getBundleSymbolicName().compareTo(o2.getBundleSymbolicName());
-				}
-			});
-
-			if (!CollectionUtils.isEqualCollection(runBundles, currentRunBundles)) {
-				printRunBundles(runBundles, System.out);
-				if (failOnChanges) {
-					throw new MojoExecutionException("The runbundles have changed. Failing the build");
-				} else {
-					logger.warn("The runbundles have changed:");
-					run.setRunBundles(runBundles);
+				String runBundles = run.resolve(failOnChanges, false);
+				run.report(errors);
+				if (errors.get() > 0) {
+					return;
 				}
+				run.setProperty(Constants.RUNBUNDLES, runBundles);
 			}
+			run.setProperty("-runbundles.tester", "biz.aQute.tester");
```

Comment:
But it is still not needed I think because biz.aQute.tester adds itself to runbundles anyway: https://github.com/bndtools/bnd/blob/master/biz.aQute.tester/src/aQute/tester/plugin/ProjectTesterImpl.java#L57